Core/include/Acts/Surfaces/Surface.hpp

```diff
@@ -407,8 +409,8 @@ class Surface : public virtual GeometryObject,
   /// @param bcheck the Boundary Check
   /// @param tolerance the tolerance used for the intersection
   ///
-  /// @return SurfaceIntersection object (contains intersection & surface)
-  virtual SurfaceIntersection intersect(
+  /// @return @c SurfaceMultiIntersection object (contains intersection & surface)
+  virtual SurfaceMultiIntersection intersect(
```

Comment:
Hmm, so **all** surfaces will now always return two intersections with up to 2 intersections, but only for cylinder (and cone?) will the second one ever be non-default.
This works, but I'm wondering if there's some way to avoid always having the extra memory footprint.

Reply:
I think this was also the case before with `alternative`